src/main/java/hydroguard/api/mock/controllers/account/AccountController.java

```diff
@@ -0,0 +1,153 @@
+            @RequestParam(required = false) LocalDateTime updatedTo,
+            @RequestParam(required = false) String sortField,
+            @RequestParam(defaultValue = "asc") String sortDirection) {
+
+        if (page < 1) {
+            throw new IllegalArgumentException("Page number must be greater than or equal to 1");
+        }
+        if (pageSize < 1 || pageSize > 100) {
+            throw new IllegalArgumentException("Page size must be between 1 and 100");
+        }
+
+        List<AccountDTO> filteredAccounts = accounts;
+
+        if (name != null && !name.isEmpty()) {
+            filteredAccounts = filteredAccounts.stream()
+                    .filter(account -> account.getName().contains(name))
+                    .collect(Collectors.toList());
+        }
+
+        if (createdFrom != null) {
+            filteredAccounts = filteredAccounts.stream()
+                    .filter(account -> !account.getCreatedAt().isBefore(createdFrom))
+                    .collect(Collectors.toList());
+        }
+
+        if (createdTo != null) {
+            filteredAccounts = filteredAccounts.stream()
+                    .filter(account -> !account.getCreatedAt().isAfter(createdTo))
+                    .collect(Collectors.toList());
+        }
+
+        if (updatedFrom != null) {
+            filteredAccounts = filteredAccounts.stream()
+                    .filter(account -> !account.getUpdatedAt().isBefore(updatedFrom))
+                    .collect(Collectors.toList());
+        }
+
+        if (updatedTo != null) {
+            filteredAccounts = filteredAccounts.stream()
+                    .filter(account -> !account.getUpdatedAt().isAfter(updatedTo))
+                    .collect(Collectors.toList());
+        }
+
+        if (sortField != null) {
+            Comparator<AccountDTO> comparator;
+            switch (sortField) {
+                case "name":
+                    comparator = Comparator.comparing(AccountDTO::getName);
+                    break;
+                case "createdAt":
+                    comparator = Comparator.comparing(AccountDTO::getCreatedAt);
+                    break;
+                case "updatedAt":
+                    comparator = Comparator.comparing(AccountDTO::getUpdatedAt);
+                    break;
+                default:
+                    throw new IllegalArgumentException("Invalid sort field");
+            }
+
+            if ("desc".equalsIgnoreCase(sortDirection)) {
+                comparator = comparator.reversed();
+            }
+
+            filteredAccounts.sort(comparator);
+        }
+
+        int total = filteredAccounts.size();
+        int start = (page - 1) * pageSize;
+        int end = Math.min(start + pageSize, total);
+
+        List<AccountDTO> paginatedAccounts = filteredAccounts.subList(start, end);
+
+        return new AccountsDTO(paginatedAccounts, total, page);
+    }
+
+    @PostMapping
```

Comment:
Post route zou wel nuttig kunnen zijn voor "Anonymous" users, maar in principe gaat het normale aanmaken van een user via keycloak. En als we hem gebruiken voor die Anonymous users dan zou er niet heel veel informatie nodig zijn in de post.